### DISCUSSION TITLE
Adding Battery

### DISCUSSION THREAD

**Discussion by CaitSith2** [sanni/cartreader#288] [Need help]
I ordered the battery module from the link listed in https://github.com/sanni/cartreader/wiki/Adding-a-battery and got a battery module that looks completely different from the module listed in the respective ebay listing picture wise.

![battery module front 2](https://user-images.githubusercontent.com/79097/141670047-8af7c74c-2cd2-4752-9150-8bccbac208de.jpg)
![battery module back 2](https://user-images.githubusercontent.com/79097/141670049-364f33fe-2b93-46b5-a3d8-928539efdf3c.jpg)

As a result,  can't really continue with the build, since these don't match what is presented in the instructions even remotely close.  I have no idea even what the slide switch even does on this module.  I know the push button switch single tap turns it on, and double-tap turns it off.

**Comment by sanni** [comment]
There are five LEDs next to the push button, I guess that is a battery level indicator?
The slide switch should turn on/off the USB Type A port.
On the old battery module the 5V und 3V pads always carried voltage and were not controlled by the switch, is this still the case?

**Comment by CaitSith2** [comment]
It seems the main thing that turns this on, is having single tapped the push button switch,  as long as the battery has sufficient charge.   I do not know if it auto shuts off, and if so, under what conditions will it not.  The 5 LEDs are in fact a battery charge indicator, but also are the power indicator as well.  If they are off, the power to the module is off, and no voltage is present on the 3v/5v outputs.  In fact, even plugging in a usb port does NOT turn on its power output.

Regardless, the module I received from that listing was "not as described" in the listing.

**Reply by Dakkaron** [reply]
It's probably a low-power standby mechanism. Those usually turn off the voltage regulator when too little power is drawn, because the voltage regulator would otherwise drain the battery in standby.

Did you try plugging a real load (e.g. charging a smartphone) into USB to see if that wakes the module up?

**Reply by CaitSith2** [reply]
Figured out the slide switch now.  It controls whether standby is enabled or not.  When it is, then there is about 2v on all of the power rails, and when you try to put a load on one of them, it turns the module completely on.  Module shuts goes into standby within 35 seconds if there is no load on its outputs.

On the other hand, if the switch is in the other position then the push button switch turns the module on and off, but haven't yet determined how long if any for no-load auto-off.  (I left it on for about 3.5 minutes before giving up on finding out.)

From the looks of things, the capacitor near the micro USB port is where I will put the 5V from arduino, if I proceed to use this module, then set the swtich to enable standby mode.  After that, the power switch on the cart dumper will be what controls if the module is on or not from that point on, based on load on the 5v rail.

**Comment by CaitSith2** [comment]
![IMG_0345](https://user-images.githubusercontent.com/79097/144708753-2249a225-69c5-46b5-acfa-c0c6c689f015.jpg)
![IMG_0347](https://user-images.githubusercontent.com/79097/144708755-50048699-9b31-41ee-bb2e-473b49ea4e09.jpg)
![IMG_0348](https://user-images.githubusercontent.com/79097/144708758-9bac40e6-5c18-4ed8-9661-00643c3ee3bf.jpg)
![IMG_0349](https://user-images.githubusercontent.com/79097/144708759-ac35f69a-c9e9-4282-a8b5-4173359723ec.jpg)

Now finally put my fully built cart dumper into a case with battery. I ended up with a battery module that needs to be turned on/off by pressing its button once/twice respectively. As a result, I went with just wire the battery module to the board directly, skip the battery module to arduino, and instead, when I need to flash the firmware, I just simply turn on the battery module and board, and left the micro-usb on the back side exposed for actual charging.

Also just confirmed that the N64 connector is in fact the nintendo standard 2.5mm pin spacing, rather than the common 2.54mm spacing, by measuring with digital calipers. This means no mis-aligned carts causing problems, in fact, I tried to mis-align a cart both to the extreme left and to the extreme right, and it still dumped correctly in both cases.

(Also there is a hole I had to put in the case to accommodate the coil that is on that battery module.  The group of 5 holes are for better seeing the charge indicator LEDs, and the hole off to the side of that was for poking a metal slotted screw driver down to the switch contacts, before I decided to mount a push button switch a little more semi-permanently.)

**Comment by majorpbx** [comment]
This thread might be of interest to you.

https://github.com/sanni/cartreader/discussions/136
